deployment/centos-package-x64/Dockerfile

```diff
@@ -1,15 +1,36 @@
 FROM centos:7
-ARG HOME=/build
-RUN mkdir /build && \
-    yum install -y @buildsys-build rpmdevtools yum-plugins-core && \
-    rpm -Uvh https://packages.microsoft.com/config/rhel/7/packages-microsoft-prod.rpm && \
-    rpmdev-setuptree
-
-WORKDIR /build/rpmbuild
-COPY ./deployment/centos-package-x64/pkg-src/jellyfin.spec SPECS
-COPY ./deployment/centos-package-x64/pkg-src/ SOURCES
-
-RUN spectool -g -R SPECS/jellyfin.spec && \
-    rpmbuild -bs SPECS/jellyfin.spec && \
-    yum-builddep  -y SRPMS/jellyfin-*.src.rpm && \
-    rpmbuild -bb SPECS/jellyfin.spec;
\ No newline at end of file
+# Docker build arguments
+ARG SOURCE_DIR=/jellyfin
+ARG PLATFORM_DIR=/jellyfin/deployment/centos-package-x64
+ARG ARTIFACT_DIR=/dist
+ARG SDK_VERSION=2.2
+# Docker run environment
+ENV SOURCE_DIR=/jellyfin
+ENV ARTIFACT_DIR=/dist
+
+# Prepare CentOS build environment
+RUN yum update -y \
+ && yum install -y @buildsys-build rpmdevtools yum-plugins-core libcurl-devel fontconfig-devel freetype-devel openssl-devel glibc-devel libicu-devel \
+ && rpm -Uvh https://packages.microsoft.com/config/rhel/7/packages-microsoft-prod.rpm \
+ && rpmdev-setuptree \
+ && yum install -y dotnet-sdk-${SDK_VERSION}
+
+# Link to docker-build script
+RUN ln -sf ${PLATFORM_DIR}/docker-build.sh /docker-build.sh
```

Comment:
Did you separate these out for the sake of commenting? If so, the comments should explain why instead of reiterating what they are doing. Ex. why instead of "x does y" or "z fails if not".

IMO - Combine all `RUN` instructions.

Reply:
Well it was to ensure that not all layers needed to be rebuild all the time, but keeping the image around makes that somewhat redundant. If it's better to combine everything I can do so.

Reply:
Completely agree that should be a goal. Adding a few `ln` and `mkdir` commands won't hurt performance :P

Reply:
Fair enough, I've combined them together. Done.